impl/src/testing.rs

```diff
@@ -56,3 +98,119 @@ pub fn make_workspace(
 pub fn make_basic_workspace() -> (TempDir, CargoWorkspaceFiles) {
   make_workspace(basic_toml(), Some(basic_lock()))
 }
+
+/**
+ * Configures the given mock_server (representing a crates.io remote) to return
+ * mock responses for the given crate and version .
+ */
+pub fn mock_remote_crate(name: &str, version: &str, mock_server: &MockServer) -> TempDir {
+  // Crate info mock response
+  mock_server.mock(|when, then| {
+    when.method(GET).path(format!("/api/v1/crates/{}", name));
+    // Note that `crate[versions]` is an arbitrary value that must only match a `versions[id]`
+    then.status(200).json_body(json!({
+        "crate": {
+            "id": name,
+            "name": name,
+            "versions": [
+                123456
+            ],
+        },
+        "versions": [
+            {
+                "id": 123456,
+                "crate": name,
+                "num": version,
+                "dl_path": format!("/api/v1/crates/{}/{}/download", name, version),
+            }
+        ],
+    }));
+  });
+
+  // Create archive contents
+  let dir = tempfile::TempDir::new().unwrap();
+  let tar_path = dir.as_ref().join(format!("{}.tar.gz", name));
+  {
+    std::fs::create_dir_all(dir.as_ref().join("archive")).unwrap();
```

Comment:
Nit: Maybe import `std::fs` and `std::fs::File` ?

Reply:
updated